### PR TITLE
chore(api): clean up api runs for local dev

### PR DESCRIPTION
**Pull request by vindard**
## Description

We were getting a port clash because admin api start was being executed twice (didn't affect functionality, but led to misleading error messages). This fixes that issue and adds dev commands for running public and admin api servers.

### CODE REVIEW

File: dev/Tiltfile [hunk 1]
```diff
@@ -177,7 +180,7 @@ local_resource(
             port = 4012,
         ),
     ),
-    deps = _buck2_dep_inputs(api_target),
+    deps = _buck2_dep_inputs(api_target_ci),
```
Comment (bodymindarts) [inline dev/Tiltfile:183]: don't see why this is required? The dev target should also know its files.

Comment (vindard) [reply dev/Tiltfile]: Fixed https://github.com/GaloyMoney/galoy/pull/3572/commits/15424efdab70847b6b0b7a298dc0d97c088f40d8

File: core/api/src/servers/graphql-admin-server.ts [hunk 2]
```diff
@@ -112,9 +112,11 @@ export async function startApolloServerForAdminSchema() {
...
-  })
-  .catch((err) => graphqlLogger.error(err, "server error"))
+if (require.main === module) {
+  setupMongoConnection()
+    .then(async () => {
+      activateLndHealthCheck()
+      await startApolloServerForAdminSchema()
+    })
+    .catch((err) => graphqlLogger.error(err, "server error"))
+}
```
Comment (nicolasburtey) [inline core/api/src/servers/graphql-admin-server.ts:122]:
```suggestion
```

Comment (nicolasburtey) [reply core/api/src/servers/graphql-admin-server.ts:122]: I think we. can just removed it

the reason I remove the wrapper around `if (require.main === module) {` initially was to make it compatible with `bun`.

File: core/api/src/servers/graphql-main-server.ts [hunk 3]
```diff
@@ -123,16 +123,18 @@ export async function startApolloServerForCoreSchema() {
...
-    ])
-  })
-  .catch((err) => baseLogger.error(err, "server error"))
+      await Promise.race([
+        startApolloServerForCoreSchema(),
+        startApolloServerForAdminSchema(),
+      ])
+    })
+    .catch((err) => baseLogger.error(err, "server error"))
+}
```
Comment (nicolasburtey) [inline core/api/src/servers/graphql-main-server.ts:140]: we should move that to another file

Comment (vindard) [reply core/api/src/servers/graphql-main-server.ts]: Done https://github.com/GaloyMoney/galoy/pull/3572/commits/7b806f1ccd581d03f639c3fe4d1af3a7cd53ef84

**Pull request merged**